### ISSUE TITLE
kube-vip as kubernetes api lb failed

### ISSUE THREAD

**Issue by FlashIvano** [plunder-app/kube-vip#69]
env file:

```bash
install -d /opt/software/env

cat > /opt/software/env/kubernetes.env <<EOF
OS_PRETTY_NAME="$(grep "PRETTY_NAME" /etc/os-release | awk -F"\"" '{print $2}')"
KUBE_VERSION="1.18.8"
KUBE_CLUSTE_NAME="mycluster"  # kubeadm的默认值为"kubernetes"

KUBE_VIP_VERSION="0.1.7"  # Get from https://github.com/plunder-app/kube-vip/releases
KUBE_VIP_INTERFACE="ens33"  # 默认路由网卡名称
KUBE_API_LB_HOST="192.168.128.250"  # 虚拟IP地址
KUBE_API_LB_PORT=16443

KUBE_POD_SUBNET="10.95.0.0/16"
KUBE_SVC_SUBNET="10.96.0.0/16"  # kubeadm默认值为"10.96.0.0/12"
EOF
```

create static pod yaml command:

```bash
source /opt/software/env/kubernetes.env


sudo docker run --network host --rm \
  plndr/kube-vip:$KUBE_VIP_VERSION \
  kubeadm init \
    --interface $KUBE_VIP_INTERFACE \
    --vip $KUBE_API_LB_HOST \
    --arp \
    --leaderElection \
  | sudo tee /etc/kubernetes/manifests/vip.yaml
```

error：
```
time="2020-08-28T02:12:48Z" level=info msg="This node is starting with leadership of the cluster"
time="2020-08-28T02:12:48Z" level=info msg="Node [k8s-node-1] is assuming leadership of the cluster"
time="2020-08-28T02:12:48Z" level=info msg="Broadcasting ARP update for 192.168.128.250 (00:0c:29:d5:81:22) via ens33"
time="2020-08-28T02:12:51Z" level=info msg="Broadcasting ARP update for 192.168.128.250 (00:0c:29:d5:81:22) via ens33"
I0828 02:12:51.937656       1 leaderelection.go:288] failed to renew lease kube-system/plunder-lock: failed to tryAcquireOrRenew context deadline exceeded
E0828 02:12:52.618251       1 leaderelection.go:307] Failed to release lock: Lease.coordination.k8s.io "plunder-lock" is invalid: spec.leaseDurationSeconds: Invalid value: 0: must be greater than 0
time="2020-08-28T02:12:52Z" level=info msg="This node is becoming a follower within the cluster"
time="2020-08-28T02:12:52Z" level=info msg="Shutting down Kube-Vip Leader Election cluster"
```

**Comment by thebsdbox** [comment]
OK, I will fix this now. I'm going to have to rip out some code from the Kubernetes code base to fix it.. we'll monitor this from the existing issue #25

**Comment by thebsdbox** [comment]
Pushed a fix into the `0.1.8` image, which hasn't been formally released yet. Please feel free to take it for a test :)

**Comment by thebsdbox** [comment]
Additionally going to add customisation of timeouts to election criteria

**Comment by thebsdbox** [comment]
Fixed with #73
